File: week01/day07/frontend/src/services/worker.ts
import { request } from './request';
import type { Worker, WorkerFormData } from '../types/worker';
import type { ApiResponse, PageResult } from '../types/api';

export function getWorkers(params?: {
  page?: number;
  pageSize?: number;
  name?: string;
  status?: string;
}): Promise<PageResult<Worker>> {
  return request<PageResult<Worker>>({
    method: 'GET',
    url: '/api/workers',
    params,
  });
}

export function getWorker(id: number): Promise<ApiResponse<Worker>> {
  return request<ApiResponse<Worker>>({
    method: 'GET',
    url: `/api/workers/${id}`,
  });
}

export function createWorker(data: WorkerFormData): Promise<ApiResponse<Worker>> {
  return request<ApiResponse<Worker>>({
    method: 'POST',
    url: '/api/workers',
    body: data,
  });
}

export function updateWorker(id: number, data: WorkerFormData): Promise<ApiResponse<Worker>> {
  return request<ApiResponse<Worker>>({
    method: 'PUT',
    url: `/api/workers/${id}`,
    body: data,
  });
}

export function updateWorkerStatus(id: number, status: string): Promise<ApiResponse<Worker>> {
  return request<ApiResponse<Worker>>({
    method: 'PATCH',
    url: `/api/workers/${id}/status`,
    body: { status },
  });
}

export function deleteWorker(id: number): Promise<ApiResponse<unknown>> {
  return request<ApiResponse<unknown>>({
    method: 'DELETE',
    url: `/api/workers/${id}`,
  });
}

export function resetPassword(id: number): Promise<ApiResponse<{ password: string }>> {
  return request<ApiResponse<{ password: string }>>({
    method: 'PATCH',
    url: `/api/workers/${id}/reset-password`,
  });
}

File: week01/day07/backend/routers/worker.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from schemas.worker import WorkerCreate, WorkerUpdate, WorkerStatusUpdate
from dependencies import get_db, get_current_admin
from services import worker as service
from utils.response import ok, ok_page

router = APIRouter(prefix="/api/workers", tags=["护工管理"])


@router.get("")
def list_workers(
    page: int = Query(1, ge=1),
    pageSize: int = Query(20, ge=1, le=100),
    name: str | None = None,
    status: str | None = None,
    db: Session = Depends(get_db),
    _=Depends(get_current_admin),
):
    result = service.list_workers(db, page=page, page_size=pageSize, name=name, status=status)
    return ok_page(result["data"], result["total"], page, pageSize)


@router.get("/{worker_id}")
def get_worker(worker_id: int, db: Session = Depends(get_db), _=Depends(get_current_admin)):
    result = service.get_worker(db, worker_id)
    return ok(result)


@router.post("")
def create_worker(data: WorkerCreate, db: Session = Depends(get_db), _=Depends(get_current_admin)):
    result = service.create_worker(db, data)
    return ok(result, "护工创建成功")


@router.put("/{worker_id}")
def update_worker(
    worker_id: int,
    data: WorkerUpdate,
    db: Session = Depends(get_db),
    _=Depends(get_current_admin),
):
    result = service.update_worker(db, worker_id, data)
    return ok(result, "更新成功")


@router.patch("/{worker_id}/status")
def update_status(
    worker_id: int,
    data: WorkerStatusUpdate,
    db: Session = Depends(get_db),
    _=Depends(get_current_admin),
):
    result = service.update_status(db, worker_id, data)
    return ok(result, "状态更新成功")


@router.delete("/{worker_id}")
def delete_worker(worker_id: int, db: Session = Depends(get_db), _=Depends(get_current_admin)):
    result = service.delete_worker(db, worker_id)
    return ok(result, "删除成功")


@router.patch("/{worker_id}/reset-password")
def reset_password(
    worker_id: int,
    db: Session = Depends(get_db),
    _=Depends(get_current_admin),
):
    result = service.reset_password(db, worker_id)
    return ok(result, "密码已重置")
